File: lib/domain/shared/ports/masters_api_credentials_resolve_output_port.rb
# frozen_string_literal: true

module Domain
  module Shared
    module Ports
      module MastersApiCredentialsResolveOutputPort
        # @param principal [Domain::Shared::Dtos::SessionPrincipal]
        def on_success(principal:)
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_invalid_api_key
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_login_required
          raise NotImplementedError, "#{self.class}##{__method__}"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/ports/masters_api_credentials_resolve_output_port.rs
use crate::shared::dtos::SessionPrincipal;

/// Ruby: `Domain::Shared::Ports::MastersApiCredentialsResolveOutputPort`
pub trait MastersApiCredentialsResolveOutputPort {
    fn on_success(&mut self, principal: SessionPrincipal);
    fn on_invalid_api_key(&mut self);
    fn on_login_required(&mut self);
}
